generators/ionic/generator.mjs

```diff
@@ -0,0 +1,458 @@
+import chalk from 'chalk';
+import { relative } from 'path';
+import _ from 'lodash';
+import { GeneratorBaseEntities, utils } from 'generator-jhipster';
+import {
+  PRIORITY_PREFIX,
+  INITIALIZING_PRIORITY,
+  CONFIGURING_PRIORITY,
+  LOADING_PRIORITY,
+  PREPARING_PRIORITY,
+  WRITING_PRIORITY,
+  WRITING_ENTITIES_PRIORITY,
+  POST_WRITING_PRIORITY,
+  INSTALL_PRIORITY,
+  END_PRIORITY,
+} from 'generator-jhipster/esm/priorities';
+
+import { DEFAULT_BACKEND_PATH, SPONSOR_MESSAGE } from '../constants.mjs';
+import { files, entityFiles } from './files.mjs';
+
+export default class extends GeneratorBaseEntities {
+  constructor(args, opts, features) {
+    super(args, opts, { taskPrefix: PRIORITY_PREFIX, ...features });
+
+    this.option('defaults', {
+      desc: 'Use default options',
+      type: String,
+    });
+
+    this.option('authentication-type', {
+      desc: 'Authentication type',
+      type: String,
+    });
+
+    this.option('base-name', {
+      desc: 'Base name',
+      type: String,
+    });
+
+    this.option('app-dir', {
+      desc: 'Directory for jhipster application',
+      type: String,
+    });
+
+    this.option('standalone', {
+      desc: 'Skip backend',
+      type: Boolean,
+    });
+
+    if (this.options.help) return;
+
+    if (this.blueprintConfig.ionicDir) {
+      throw new Error('Ionic generator must run at ionic application folder');
+    }
+
+    this.ionicStorage = this.blueprintStorage;
+    this.ionicConfig = this.blueprintConfig;
+
+    this.localJHipsterStorage = this._config;
+    this.localJHipsterConfig = this.jhipsterConfig;
+
+    if (this.options.standalone) {
+      this.ionicConfig.appDir = false;
+    }
+    if (this.options.appDir !== undefined) {
+      this.ionicConfig.appDir = this.options.appDir;
+    }
+    if (this.options.baseName !== undefined) {
+      this.localJHipsterConfig.baseName = this.options.baseName;
+    }
+    if (this.options.authenticationType !== undefined) {
+      this.localJHipsterConfig.authenticationType = this.options.authenticationType;
+    }
+    if (this.options.defaults || this.options.force) {
+      this.ionicStorage.defaults({ appDir: DEFAULT_BACKEND_PATH });
+    }
+  }
+
+  async _postConstruct() {
+    await this.prompt(
+      [
+        {
+          type: 'input',
+          name: 'appDir',
+          message: 'Where JHipster backend application is located?',
```

Comment:
```suggestion
          message: 'Enter the directory where your JHipster app is located:',
```
To match previous version so docs don't need updating.

Reply:
Where is the `What is the project name of your application?` prompt defined? I'd like to change it to match the previous: `What do you want to name your Ionic application?`

Reply:
And how do we suppress the "⬢ Welcome to the JHipster Project Name ⬢" message?

Reply:
I am trying to reuse basic generators, I will change at generator-jhipster side.

Reply:
Does that mean this will require a new release of JHipster for these changes?

Reply:
Should be fixed.